pkg/providers/ovirt/client/ovirt-client.go

```diff
@@ -0,0 +1,270 @@
+package ovirtclient
+
+import (
+	"fmt"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+// OvirtClient retrieves rich VM object - with all necessary links have been followed
+type OvirtClient interface {
+	GetVM(connectionSettings *ConnectionSettings, id *string, name *string, cluster *string) (*ovirtsdk.Vm, error)
+}
+
+// ConnectionSettings wrap information required to make oVirt API connection
+type ConnectionSettings struct {
+	URL      string
+	Username string
+	Password string
+	Insecure bool
+}
+
+// RichOvirtClient is responsilbe for retrieving VM data from oVirt API
+type RichOvirtClient struct {
```

Comment:
Why not to keep the connection in the struct?

Reply:
In that case, I think, we would have to expose con.Close() as another function of that API and add probably some logic for checking if it's connected or not.
I see more value in moving `ConnectionSettings` from the method arguments to the struct and connecting/disconnecting in the retrieval logic as it is. What do you think?

Reply:
Reconnecting takes time like handshaking and creating http session. Please check any examples from [here](https://github.com/oVirt/ovirt-engine-sdk-go/tree/master/sdk/examples) to see how complex is Close().

Reply:
OK. Is your assumption that this client will have more functionality (like starting, stopping vm, updating name/etc. to flag it's been migrated) and we want to connect only once?

Reply:
I have just pushed the client with struct-bound connection. Is that what you had in mind?

Reply:
Yes :+1: